bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/handler/AbstractNukiDeviceHandler.java

```diff
@@ -0,0 +1,320 @@
+import org.openhab.binding.nuki.internal.dataexchange.NukiBaseResponse;
+import org.openhab.binding.nuki.internal.dataexchange.NukiHttpClient;
+import org.openhab.binding.nuki.internal.dto.BridgeApiDeviceStateDto;
+import org.openhab.binding.nuki.internal.dto.BridgeApiListDeviceDto;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AbstractNukiDeviceHandler} is a base class for implementing ThingHandlers for Nuki devices
+ *
+ * @author Jan Vybíral - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractNukiDeviceHandler extends BaseThingHandler {
+
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final int JOB_INTERVAL = 60;
+
+    @Nullable
+    private NukiHttpClient nukiHttpClient;
+    @Nullable
+    protected ScheduledFuture<?> reInitJob;
+    protected final String nukiId;
+
+    public AbstractNukiDeviceHandler(Thing thing) {
+        super(thing);
+        logger.debug("Instantiating {}({})", getClass().getSimpleName(), thing);
+        String id = thing.getProperties().get(NukiBindingConstants.PROPERTY_NUKI_ID);
+        if (id == null) {
+            Object idFromOldConfig = getConfig().get(NukiBindingConstants.PROPERTY_NUKI_ID);
+            if (idFromOldConfig != null) {
+                logger.warn(
+                        "SmartLock '{}' was created by old version of binding. It is recommended to delete it and discover again",
+                        thing.getUID());
+                int nukiId = Integer.parseInt(idFromOldConfig.toString(), 16);
+                this.nukiId = Integer.toString(nukiId);
+            } else {
+                throw new IllegalStateException(String.format(
+                        "%s is missing from properties of %s. Delete thing and add it again using discovery",
+                        NukiBindingConstants.PROPERTY_NUKI_ID, thing));
+            }
+        } else {
+            nukiId = id;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initialize() for Nuki Device[{}].", getThing().getUID());
+        scheduler.execute(this::initializeHandler);
```

Comment:
During init, you could set the thing status to UNKNOWN.

Reply:
Any reason why? The thing is in INITIALIZING state which seems more appropriate

Reply:
I don't know why I commented that. You can leave it as is.